bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -266,15 +266,42 @@ def get_headnode(backend: BaseBackend, npartitions: int, *args) -> HeadNode:
     if isinstance(firstarg, int):
         # RDataFrame(ULong64_t numEntries)
         return EmptySourceHeadNode(backend, npartitions, localdf, firstarg)
-    elif isinstance(firstarg, (ROOT.TTree, str)):
-        # RDataFrame(std::string_view treeName, filenameglob, defaultBranches = {})
-        # RDataFrame(std::string_view treename, filenames, defaultBranches = {})
-        # RDataFrame(std::string_view treeName, dirPtr, defaultBranches = {})
-        # RDataFrame(TTree &tree, const ColumnNames_t &defaultBranches = {})
-        return TreeHeadNode(backend, npartitions, localdf, *args)
+    elif isinstance(firstarg, (ROOT.TTree)):
+    #     # RDataFrame(std::string_view treeName, filenameglob, defaultBranches = {})
+    #     # RDataFrame(std::string_view treename, filenames, defaultBranches = {})
+    #     # RDataFrame(std::string_view treeName, dirPtr, defaultBranches = {})
+    #     # RDataFrame(TTree &tree, const ColumnNames_t &defaultBranches = {})
+        return TreeHeadNode(backend, npartitions, localdf, *args) 
+    elif isinstance(firstarg, str):
+        # if first argument is a string, we want to check if the second argument
+        # which is a ROOT file holds a TTree or RNTuple
+        secondarg = args[1]
+ 
+        if (isinstance (secondarg, (str, ROOT.std.string, ROOT.std.string_view))):
+            
+            wildcards = ["[", "]", "*", "?"]
+            if (any(wildcard in secondarg for wildcard in wildcards)):
+                inFile =  ROOT.TFile.Open(ROOT.Internal.TreeUtils.ExpandGlob(secondarg)[0])
+            else:
+                inFile =  ROOT.TFile.Open(secondarg) 
+
+        else:
+            inFile =  ROOT.TFile.Open(secondarg[0])
+
+        inFileCheck = inFile.Get(firstarg)
+
+        if isinstance(inFileCheck, ROOT.TTree):
+            return TreeHeadNode(backend, npartitions, localdf, *args) 
+            
+        elif isinstance(inFileCheck, ROOT.Experimental.RNTuple):
+            return RNTupleHeadNode(backend, npartitions, localdf, *args)
+            
+        else:
+            raise RuntimeError("Something is wrong with new constructor")
```

Comment:
I suspect the cause of the current failure is the fact we call `TFile::Open` without the proper guards and that sets the global ROOT directory (i.e. `gDirectory`) so in tests where we write objects to disk those are actually being written to the wrong place. I think this is the safest way we can approach this:
1. figure out at first the correct path to open, i.e. storing a variable e.g. `path` with either `secondarg` or the right sanitized string
2. Open the file in a with context with a TContext as such
```python
with TDirectory.TContext(), TFile.Open(path, "READ_WITHOUT_GLOBALREGISTRATION") as f:
    dataset = f.Get(firstarg)
    # Continue with checking the type of the dataset
```

For a longer explanation on what the code above does, see [this tutorial](https://root.cern/doc/master/pyroot006__tcontext__context__manager_8py.html) and [TContext doc](https://root.cern.ch/doc/master/classTDirectory_1_1TContext.html#autotoc_md1422)